### PR TITLE
Allow using a fork of the Uyuni project

### PR DESCRIPTION
**Pull request by cbosdo**

### CODE REVIEW

**Comment by cbosdo** [comment]
@juliogonzalez I think you'ld be interested in this one PR

**Comment by Bischoff** [comment]
@cbosdo real cool, we need that for so much time.

Can you add it to other modules too please? It's just copy and paste. If not I can do it.

**Comment by cbosdo** [comment]
@juliogonzalez @Bischoff :heavy_check_mark:

File: README_ADVANCED.md [hunk 1]
```diff
@@ -673,3 +673,14 @@ module "sumamail3" {
   traceback_email = "michele.bologna@chameleon-mail.com"
 }
 ```
+
+## Testing a fork
+
+With the default configuration, either the spacewalk or the Uyuni repository is cloned in the controller.
```
Comment (juliogonzalez) [inline README_ADVANCED.md]: I'd improve this a little:
```
With the default configuration, either the spacewalk or the Uyuni repository is cloned in the controller according to the selected branch (Manager uses Uyuni, all others use spacewalk)
```

Comment (cbosdo) [reply README_ADVANCED.md]: done

**Comment by juliogonzalez** [comment]
FMPOV you can mege :-D

File: README_ADVANCED.md [hunk 2]
```diff
@@ -673,3 +673,15 @@ module "sumamail3" {
   traceback_email = "michele.bologna@chameleon-mail.com"
 }
 ```
+
+## Testing a fork
+
+With the default configuration, either the spacewalk or the Uyuni repository is cloned in the controller
+according to the selected branch (Manager uses Uyuni, all others use spacewalk).
+Since it is recommended to fork the Uyuni repository to create new features, testing against a branch of
```
Comment (moio) [inline README_ADVANCED.md]: I think it is not very clear from this description that what this patch does is to allow _the testsuite code_ to be checked from a specified fork, while the _product code_ will remain the one as specified via `product_version`.

We need a different mechanism to test _products_ from forks, and that is basically having RPMs built per PR. This is something that will hopefully happen soon, it's just not there at the moment.

Please move this section as a subsection of "Cucumber testsuite"

Comment (cbosdo) [reply README_ADVANCED.md]: How could RPMs help when the git repo is only used for the testsuite code?

Comment (moio) [reply README_ADVANCED.md]: The product is currently installed via RPMs, the testsuite code via git clone.

First problem: the testsuite is not packaged, it should probably be. In that case we wouldn't have to have this PR.

Second problem: we presently don't have a way to run the _products_ (particularly: the SUSE Manager Server) off a git clone and that will probably never happen, due to the complicated build process. Thus we would need to build fork-specific/PR-specific RPMs in order to have fork-specific/PR-specific sumaformed VMs.

**Comment by Bischoff** [comment]
@cbosdo Could you also update the templates, please? Sorry :blush:

File: README_ADVANCED.md (hunk 2 above)
Comment (cbosdo) [reply README_ADVANCED.md]: I agree, however we need at least a temporary workaround for those problems.

Comment (Bischoff) [reply README_ADVANCED.md]: @moio I disagree with a packaged testsuite. How would we test our experimental git branches? I don't want the burden of packaging them...

Even if a packaged testsuite would make sense _per se_, I still want the clone approach in the suite.

**Comment by cbosdo** [comment]
Moved the doc to the testsuite section.

**Comment by cbosdo** [comment]
@Bischoff done

**Comment by Bischoff** [comment]
Thank you for your understanding, @moio.

**Pull request merged**